tests/telemetry/test_track.py

```diff
@@ -136,38 +144,45 @@ def mock_requests_get(*args, **kwargs):
         )
 
     with mock.patch("mlflow.telemetry.client.requests.get", side_effect=mock_requests_get):
-        set_telemetry_client()
-        client = get_telemetry_client()
+        client = TelemetryClient()
         assert client is not None
         assert client.config is None
-        # do not skip when config is not fetched yet
-        assert _is_telemetry_disabled_for_event(TestEvent) is False
-        assert _is_telemetry_disabled_for_event(TestEvent) is False
-        time.sleep(2)
-        assert client._is_config_fetched is True
-        assert client.config is not None
-        # event not in disable_events, do not skip
-        assert _is_telemetry_disabled_for_event(CreateLoggedModelEvent) is False
-        # event in disable_events, skip
-        assert _is_telemetry_disabled_for_event(TestEvent) is True
+        with mock.patch("mlflow.telemetry.track.get_telemetry_client", return_value=client):
+            # do not skip when config is not fetched yet
+            assert _is_telemetry_disabled_for_event(TestEvent) is False
+            assert _is_telemetry_disabled_for_event(TestEvent) is False
+            time.sleep(2)
+            assert client._is_config_fetched is True
+            assert client.config is not None
+            # event not in disable_events, do not skip
+            assert _is_telemetry_disabled_for_event(CreateLoggedModelEvent) is False
+            # event in disable_events, skip
+            assert _is_telemetry_disabled_for_event(TestEvent) is True
+        # clean up
+        client._clean_up()
```

Comment:
Can we make `TelemetryClient` a context manager to ensure `_clean_up` is always called.

Reply:
I think we can create a context manager for test-only purpose, the client itself shouldn't be a context manager. 
But can I fix this later? I should have more time refactoring the tests later after the release